src/metabase/util/time/impl.cljs

```diff
@@ -151,7 +151,7 @@
   (.local value))
 
 (def ^:private parse-time-formats
-  #js ["HH:mm:ss.SSS[Z]"
```

Comment:
On FE we use `SSS[Z]` in many cases, especially in viz. It's unclear whether 1) this is a correct fix 2) if yes, it should probably be updated everywhere.

Reply:
@metamben could you please clarify for a frontend engineer what are the implications of these changes to what parts of the code? Frontend does use `[Z]` format in a few places but I do not understand if the PR changes affect these places. There are two specific for ECharts rendering which won't be affected as long as the data formats in datasets remain the same. And there is one usage in `frontend/src/metabase/lib/time.ts` that is used by formatting

Reply:
My change only affects parsing. My understanding is that when parsing with `[Z]` in the format, moment doesn't interpret the trailing `Z` as a timezone marker, and it parses as a local (date)time. My experience is that when formatting, the (date)times are formatted in `GMT` and `[Z]` adds the trailing `Z`.

The effect of the change is that Metabase lib should now parse (date)times with a trailing `Z` in the `GMT` timezone (as opposed to the local timezone).

> It's unclear whether 1) this is a correct fix

What is needed to convince ourselves? What are alternatives?

> 2) if yes, it should probably be updated everywhere.

My guess is that parsing with `[Z]` hardly does what was intended and should be updated.

Reply:
Understood, thank you! So now I'm confident this change won't affect the presentation layer. But I don't think I can meaningfully approve PR by myself as I'm not familiar with the area it actually impacts